src/tests/adding.md

```diff
@@ -0,0 +1,273 @@
+# Adding new tests
+
+**In general, we expect every PR that fixes a bug in rustc to come
+accompanied with a regression test of some kind.** This test should
+fail in master but pass after the PR. These tests are really useful
+for preventing us from repeating the mistakes of the past.
+
+To add a new test, the first thing you generally do is to create a
+file, typically a Rust source file. Test files have a particular
+structure:
+
+- They always begin with the copyright notice;
```

Comment:
Maybe it would be useful to link the copyright notice here!

Reply:
ugh. I just wish it didn't exist. But I suppose you are right.

Reply:
@nikomatsakis I totally understand your pain